internal/controllers/topology/cluster/desired_state_test.go

```diff
@@ -1718,75 +1712,53 @@ func TestComputeMachineDeployment(t *testing.T) {
 }
 
 func TestComputeMachineDeploymentVersion(t *testing.T) {
-	controlPlaneStable122 := builder.ControlPlane("test1", "cp1").
-		WithSpecFields(map[string]interface{}{
-			"spec.version":  "v1.2.2",
-			"spec.replicas": int64(2),
-		}).
-		WithStatusFields(map[string]interface{}{
-			"status.version":             "v1.2.2",
-			"status.replicas":            int64(2),
-			"status.updatedReplicas":     int64(2),
-			"status.readyReplicas":       int64(2),
-			"status.unavailableReplicas": int64(0),
-		}).
-		Build()
-	controlPlaneStable123 := builder.ControlPlane("test1", "cp1").
-		WithSpecFields(map[string]interface{}{
-			"spec.version":  "v1.2.3",
-			"spec.replicas": int64(2),
-		}).
-		WithStatusFields(map[string]interface{}{
-			"status.version":             "v1.2.3",
-			"status.replicas":            int64(2),
-			"status.updatedReplicas":     int64(2),
-			"status.readyReplicas":       int64(2),
-			"status.unavailableReplicas": int64(0),
-		}).
-		Build()
-	controlPlaneUpgrading := builder.ControlPlane("test1", "cp1").
-		WithSpecFields(map[string]interface{}{
-			"spec.version": "v1.2.3",
-		}).
-		WithStatusFields(map[string]interface{}{
-			"status.version": "v1.2.1",
-		}).
-		Build()
-	controlPlaneScaling := builder.ControlPlane("test1", "cp1").
-		WithSpecFields(map[string]interface{}{
-			"spec.version":  "v1.2.3",
-			"spec.replicas": int64(2),
-		}).
-		WithStatusFields(map[string]interface{}{
-			"status.version":             "v1.2.3",
-			"status.replicas":            int64(1),
-			"status.updatedReplicas":     int64(1),
-			"status.readyReplicas":       int64(1),
-			"status.unavailableReplicas": int64(0),
-		}).
-		Build()
-	controlPlaneDesired := builder.ControlPlane("test1", "cp1").
-		WithSpecFields(map[string]interface{}{
-			"spec.version": "v1.2.3",
-		}).
+	controlPlaneObj := builder.ControlPlane("test1", "cp1").
 		Build()
 
+	mdName := "md-1"
+	currentMachineDeploymentState := &scope.MachineDeploymentState{Object: builder.MachineDeployment("test1", mdName).WithVersion("v1.2.2").Build()}
+
 	tests := []struct {
-		name                          string
-		machineDeploymentTopology     clusterv1.MachineDeploymentTopology
-		currentMachineDeploymentState *scope.MachineDeploymentState
-		upgradingMachineDeployments   []string
-		upgradeConcurrency            int
-		currentControlPlane           *unstructured.Unstructured
-		desiredControlPlane           *unstructured.Unstructured
-		topologyVersion               string
-		expectedVersion               string
+		name                                 string
+		machineDeploymentTopology            clusterv1.MachineDeploymentTopology
+		currentMachineDeploymentState        *scope.MachineDeploymentState
+		upgradingMachineDeployments          []string
+		upgradeConcurrency                   int
+		controlPlaneStartingUpgrade          bool
```

Comment:
We dont need this for the current tests because `controlPlanePendingUpgrade` is essentially covered by all other cases. `IsPendingUpgrade` is basically a result of all other checks combined and since in these tests we have all the individual checks they are already extensively covering all the cases.